h2o-py/h2o/explanation/_explain.py

```diff
@@ -2722,6 +2739,14 @@ def explain_row(
             warnings.warn("No model with variable importance. Selecting all features to explain.")
             columns_of_interest = _get_xy(models_to_show[0])[0]
 
+    # Make sure that there are no string columns to explain as they are not supported by pdp
+    # Usually those columns would not be used by algos so this just makes sure to exclude them
+    # if user specifies top_n=float('inf') or columns_of_interest=x etc.
+    dropped_string_columns = [col for col in columns_of_interest if frame.type(col) == "string"]
+    if len(dropped_string_columns) > 0:
+        warnings.warn("Dropping string columns as they are not supported: {}".format(dropped_string_columns))
+        columns_of_interest = [col for col in columns_of_interest if frame.type(col) != "string"]
```

Comment:
*fancy*: if you  want to avoid the double iteration, one of my favourite utility fn:
```python
def partition(iterable, predicate=bool):
    iter1, iter2 = tee((predicate(it), it) for it in iterable)
    return (it for b, it in iter1 if b), (it for b, it in iter2 if not b)
```

Reply:
Fancy indeed.

But I'd rather keep it as it is - explicit and clear until we drop support for py2.7 - since `filter` or `map` is eager there I  had to check if `tee` is eager or not and I'd prefer source code to be "simpler". Also the double iteration shouldn't be a performance issue here since the types are cached in the client.